### PR TITLE
feat(lanelet2_extension): added localization landmark

### PR DESCRIPTION
**Pull request by SakodaShintaro**
## Description

Added localization landmark.

This new lanelet2 definition is related to the `landmark_based_localizer` in autoware.universe.

<https://github.com/autowarefoundation/autoware.universe/tree/main/localization/landmark_based_localizer>

Related pull request https://github.com/autowarefoundation/autoware.universe/pull/5803

## Tests performed

It has been confirmed that the `logging_simulator` with `pose_source:=artag` runs with the same accuracy as before on AWSIM data with GT.

## Effects on system behavior

There is no effect except for `ar_tag_based_localizer`, and there is no change in the behavior of `ar_tag_based_localizer`. Just moved the definition code.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by SakodaShintaro** [comment]
@mitsudome-r friendly ping.

File: tmp/lanelet2_extension/include/lanelet2_extension/localization/landmark.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace lanelet::localization
+{
+
+struct Landmark
+{
+  std::string id;
+  geometry_msgs::msg::Pose pose;
+};
+
+std::vector<Landmark> parseLandmarks(
```
[the remark this answers is not in the document]
Comment (SakodaShintaro) [reply tmp/lanelet2_extension/include/lanelet2_extension/localization/landmark.hpp]: @mitsudome-r 
Thank you for your comment.
Approach 1 was adopted. Place only the first half of the Polygon parsing function here, and place the second half of the landmark analysis function in autoware.universe.
Fixed in https://github.com/autowarefoundation/autoware_common/pull/219/commits/db4a9d4b99a70eacb6eafdb61c19e68a1c2fa1bf

File: tmp/lanelet2_extension/include/lanelet2_extension/localization/landmark.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <lanelet2_core/primitives/Polygon.h>
+
+#include <string>
+#include <vector>
+
+namespace lanelet::localization
+{
+
+std::vector<lanelet::Polygon3d> parseLandmarkPolygons(
```
[the remark this answers is not in the document]
Comment (SakodaShintaro) [reply tmp/lanelet2_extension/include/lanelet2_extension/localization/landmark.hpp:28]: @mitsudome-r 
The other code in lanelet2_extentions usually uses camel case, so I did that in this code as well, but clang-tidy seems to give a warning. Should I fix it?

[the remark this answers is not in the document]
Comment (mitsudome-r) [reply tmp/lanelet2_extension/include/lanelet2_extension/localization/landmark.hpp:28]: I think we can keep it constant with the other functions.
If needed, we can change all at once.

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**